src/actions/products.js

```diff
@@ -0,0 +1,49 @@
+import { requestGetProductList, requestGetProduct } from 'api';
+import { 비동기_요청 } from 'constants/';
+import { 상품리스트_불러오기_액션, 상품_불러오기_액션 } from './types';
+
+const getProductList = () => async (dispatch) => {
+  dispatch({
+    type: 상품리스트_불러오기_액션.PENDING,
+  });
+
+  const response = await requestGetProductList();
+
+  if (response.status === 비동기_요청.FAILURE) {
+    dispatch({
+      type: 상품리스트_불러오기_액션.FAILURE,
+      payload: response.content,
+    });
+
+    return;
+  }
+
+  dispatch({
+    type: 상품리스트_불러오기_액션.SUCCESS,
+    payload: response.content,
+  });
+};
```

Comment:
액션들 전부 마찬가지이지만 if 문을 추가하지 않고 삼항 연산자로 나누어야할 부분만 나눠보는게 어떨까요?

```suggestion
const actionType = response.status === 비동기_요청.FAILURE 
  ? 상품리스트_불러오기_액션.FAILURE
  : 상품리스트_불러오기_액션.SUCCESS;
  
dispatch({
  type: actionType,
  payload: response.content,
});
```

Reply:
헉.. early return 형태만이 능사가 아니었네요!
삼항 연산자를 사용해서 나눠주고, 중복이 발생해서 유틸 함수로 분리해주었습니다!

[refactor : action dispatch 로직 유틸 함수로 분리](https://github.com/usageness/react-shopping-cart-prod/commit/13d2a2a26893c5fc9de56ef5c841f4522a130ac3)

```js
// utils/asyncDispatchAction.js
const asyncDispatchAction = (dispatch, response, action) => {
  const actionType = response.status === 비동기_요청.FAILURE ? action.FAILURE : action.SUCCESS;

  dispatch({
    type: actionType,
    payload: response.content,
  });
};
```